File: stack/dailyTemperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        res = [0] * len(temperatures)
        for i, temp in enumerate(temperatures):
            while len(stack) and temp > stack[-1][0]:
                update = stack.pop()
                res[update[1]] = i - update[1]
            stack.append([temp, i])
        return res

File: stack/dailyTemperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> res(temperatures.size(), 0);
        vector<pair<int, int>> stack;

        for(int i = 0; i < temperatures.size(); i++){
            int cur_temp = temperatures[i];

            while(stack.size() && cur_temp > stack.back().first){
                res[stack.back().second] = i - stack.back().second;
                stack.pop_back(); 
            }
            stack.push_back({cur_temp, i});
        }
        return res;
    }
};
